tfs/src/main/java/hudson/plugins/tfs/rm/ConnectReleaseWebHookEvent.java

```diff
@@ -0,0 +1,319 @@
+    private static final String CREATE_WEBHOOK_EVENTNAME = "rmwebhook-create";
+    private static final String REMOVE_WEBHOOK_ENENTNAME = "rmwebhook-remove";
+    private static final String LIST_WEBHOOK_EVENTNAME = "rmwebhook-list";
+    private static final String LINK_WEBHOOK_EVENTNAME = "rmwebhook-link";
+    private static final String UNLINK_WEBHOOK_EVENTNAME = "rmwebhook-unlink";
+
+    /**
+     * Factory to create ConnectReleaseWebHookEvent.
+     */
+    public static class Factory implements AbstractHookEvent.Factory {
+        @Override
+        public ConnectReleaseWebHookEvent create() {
+            return new ConnectReleaseWebHookEvent();
+        }
+
+        @Override
+        public String getSampleRequestPayload() {
+            return "{\n"
+                 + "    \"eventType\": rmwebhook-create\n"
+                 + "    \"resource\": {"
+                 + "       \"webhookName\": \"webhook name\"\n"
+                 + "       \"payloadUrl\": \"https://xplatalm.vsrm.visualstudio.com/_apis/Release/receiveExternalEvent/wenhookId\"\n"
+                 + "       \"secret\": \"secret\"\n"
+                 + "     }"
+                 + "}";
+        }
+    }
+
+    @Override
+    public JSONObject perform(final ObjectMapper mapper, final Event event, final String message, final String detailedMessage) {
+        final Object resource = event.getResource();
+        final ReleaseWebHookResource parameters = mapper.convertValue(resource, ReleaseWebHookResource.class);
+
+        if (event.getEventType().equalsIgnoreCase(CREATE_WEBHOOK_EVENTNAME)) {
+            createReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(REMOVE_WEBHOOK_ENENTNAME)) {
+            deleteReleaseWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(LIST_WEBHOOK_EVENTNAME)) {
+            return listReleaseWebHook();
+        } else if (event.getEventType().equalsIgnoreCase(LINK_WEBHOOK_EVENTNAME)) {
+            linkWebHook(parameters);
+        } else if (event.getEventType().equalsIgnoreCase(UNLINK_WEBHOOK_EVENTNAME)) {
+            unlinkWebHook(parameters);
+        } else {
+            throw new UnsupportedOperationException("Webhook operation " + parameters.getOperationType() + " is not supported");
+        }
+
+        return JSONObject.fromObject(event);
+    }
+
+    private String validateAndGetPayloadUrl(final ReleaseWebHookResource parameters) {
+        String payloadUrl = parameters.getPayloadUrl();
+
+        if (StringUtils.isBlank(payloadUrl)) {
+            throw new InvalidParameterException("pyaloadUrl is empty");
+        }
+
+        final URI uri;
+        try {
+            uri = new URI(payloadUrl);
+        } catch (final URISyntaxException e) {
+            throw new InvalidParameterException("Malformed Payload URL " + e.getMessage());
+        }
+
+        final String hostName = uri.getHost();
+        if (StringUtils.isBlank(hostName)) {
```

Comment:
how is isBlank different from isEmpty? should we use one of these across all instances?

Reply:
IsBlank is what we should use. IsBlank is equals to String.IsNullOtEmpty. Changed all instance to IsBlank

Reply:
Still see couple of instances, pl. check...